### PR TITLE
enlightenment.efl: 1.22.5 -> 1.23.1

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- enlightenment.efl: 1.22.5 -> [1.23.1](https://www.enlightenment.org/news/efl-1.23.1)
- pythonPackage.python-efl: 1.22.0 -> [1.23.0](https://sourceforge.net/p/enlightenment/mailman/message/36779150/)
- enlightenment.{enlightenment,ephoto,rage,terminology}: use default pkgconfig
- enlightenment.{econnman,efl,terminology}, pythonPackages.python-efl: add romildo as maintainer
- _Edited_: enlightenment.efl: remove dependence on xineLib 

Tested on a VM.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @matejc @ftrvxmtrx

### CODE REVIEW

**Comment by romildo** [comment]
@GrahamcOfBorg build enlightenment.efl python2Packages.pythonefl enlightenment.econnman enlightenment.ephoto enlightenment.rage enlightenment.terminology enlightenment.enlightenment

**Review by c0bw3b — APPROVED** [review]
Result of `nix-review pr 72027` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>9 package were build:</summary>

  - enlightenment.econnman
  - enlightenment.efl
  - enlightenment.enlightenment
  - enlightenment.ephoto
  - enlightenment.rage
  - enlightenment.terminology
  - python27Packages.pythonefl
  - python37Packages.pythonefl
  - python38Packages.pythonefl
</details>

File: pkgs/desktops/enlightenment/efl.nix [hunk 1]
```diff
@@ -90,30 +94,25 @@ stdenv.mkDerivation rec {
...
-    "--enable-liblz4"
-    "--enable-systemd"
-    "--enable-image-loader-webp"
-    "--enable-harfbuzz"
-    "--enable-xine"
-    "--enable-fb"
-    "--disable-tslib"
-    "--with-systemdunitdir=$out/systemd/user"
-    "ac_ct_CXX=foo"
+  mesonFlags = [
```
Comment (c0bw3b) [inline pkgs/desktops/enlightenment/efl.nix]: I guess the other flags corresponding to former `configure` flags are ON by default or auto-detected?
Things like `harfbuzz`, `systemd`, `elput` ..

Comment (romildo) [reply pkgs/desktops/enlightenment/efl.nix]: Except `--enable-fb`, which defaults to false.

Comment (romildo) [reply pkgs/desktops/enlightenment/efl.nix]: [`meson-options.txt`](https://git.enlightenment.org/core/efl.git/tree/meson_options.txt) does not mention ` --enable-elput` abd `--enable-multisense`. According to raster (on IRC) they do not exist anymore and  they are on by default, or they are implied by some other config option.

Comment (c0bw3b) [reply pkgs/desktops/enlightenment/efl.nix]: :+1: thanks for the research

**Pull request merged**